### PR TITLE
expose keyboardType in DropdownMenu #150894

### LINKED ISSUE #150894: DropdownMenu allow setting keyboard type
### Use case

When using the DropdownMenu widget, I need to be able to set the keyboard type that appears.

### Proposal

DropdownMenu should expose a parameter to set the `keyboardType` of the underlying text field.

### PR DESCRIPTION
**Pull request by j7126**
This PR exposes the `keyboardType` parameter in DropdownMenu, allowing the type of keyboard to be used for the text input to be changed.

fixes #150894

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by Piinks** [comment]
Thank you for your contribution. This PR will need tests before we can review it. Thanks! If you are not sure how to write tests, the #hackers-new channel on our Discord is available for new contributors seeking guidance.

**Review by QuncCccccc — APPROVED** [review]
LGTM:) Thanks for the contribution!

File: packages/flutter/lib/src/material/dropdown_menu.dart [hunk 1]
```diff
@@ -266,6 +267,11 @@ class DropdownMenu<T> extends StatefulWidget {
   /// Defaults to true as the search function could be commonly used.
   final bool enableSearch;
 
+  /// The type of keyboard to use for editing the text.
+  ///
+  /// Defaults to [TextInputType.text].
```
Comment (justinmc) [inline packages/flutter/lib/src/material/dropdown_menu.dart:272]: Nit: Maybe enforce this by setting a real default at line 162? Up to you.

Comment (justinmc) [reply packages/flutter/lib/src/material/dropdown_menu.dart:272]: Or if you don't do this, add a test that verifies the default value that textField.keyboardType ends up with when keyboardType is not passed in DropdownMenu.

**Review by justinmc — APPROVED** [review]
LGTM with a nit :+1:

File: packages/flutter/lib/src/material/dropdown_menu.dart (hunk 1 above)
Comment (j7126) [reply packages/flutter/lib/src/material/dropdown_menu.dart:272]: I have added the additional test for keyboardtype default

**Pull request merged**